packages/experiments/src/components/CollapsibleSection/CollapsibleSection.view.tsx

```diff
@@ -23,16 +24,71 @@ export const CollapsibleSectionView = (
 ) => {
   const { collapsed, titleAs: TitleType = CollapsibleSectionTitle, titleProps, children } = props;
 
-  // TODO: make sure onToggleCollapse gets called both from state and from titleProps.
+  const onRootKeyDown = (ev: React.KeyboardEvent<Element>) => {
+    const rootKey = getRTL() ? KeyCodes.right : KeyCodes.left;
+    switch (ev.which) {
+      case rootKey:
+        if (props.titleElementRef && props.titleElementRef.value && ev.target !== props.titleElementRef.value) {
+          props.titleElementRef.value.focus();
+          ev.preventDefault();
+          ev.stopPropagation();
+        }
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const onToggleCollapse = () => {
+    if (props.titleProps && props.titleProps.onToggleCollapse) {
+      props.titleProps.onToggleCollapse();
+    }
+    if (props.onToggleCollapse) {
+      props.onToggleCollapse();
+    }
+  };
+
+  const onKeyDown = (ev: React.KeyboardEvent<Element>) => {
+    const collapseKey = getRTL() ? KeyCodes.right : KeyCodes.left;
+    const expandKey = getRTL() ? KeyCodes.left : KeyCodes.right;
+
+    switch (ev.which) {
+      case collapseKey:
+        if (!collapsed) {
+          if (onToggleCollapse) {
+            onToggleCollapse();
+          }
+          break;
+        }
+        return;
+
+      case expandKey:
+        if (collapsed) {
+          if (onToggleCollapse) {
+            onToggleCollapse();
+          }
+          break;
+        }
+        return;
+
+      default:
+        return;
+    }
```

Comment:
I'm going back and forth a bit about whether these handlers should be in View or State. I can see the argument that their concerns of View in translating user input to View callbacks, but then I also feel Views should be very simple and only concerned with rendering output. Anybody have any input?

Reply:
oh i see this comment after commenting above! I think I believe the later, that views should be really simple and only concerned with rendering output.

Reply:
I tend to agree. I'll move it back to state.